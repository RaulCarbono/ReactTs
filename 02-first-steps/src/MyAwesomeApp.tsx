export function MyAwesomeApp() {

    const firstName = 'Raul';
    const lastName = 'Carbono'


    return (
        <div>
        <h3 className="estoy loco">{lastName}</h3>    
        <h1 data-testid="first-name-title">{firstName}!!</h1>
        </div>
    )
}