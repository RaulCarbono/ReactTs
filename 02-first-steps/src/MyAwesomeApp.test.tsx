import { describe, expect, test } from "vitest";
import { render, screen } from '@testing-library/react'

import { MyAwesomeApp } from "./MyAwesomeApp";

describe('MyAwesomeApp', () => {
     test('Should render firstName and lastname', () => {
       const {container} = render(<MyAwesomeApp/>)
       screen.debug();

      //  console.log(container.innerHTML)
      const h1 = container.querySelector('h1')
      const h3 = container.querySelector('h3')

       expect(h1?.innerHTML).toContain('Raul')
       expect(h3?.innerHTML).toContain('Carbono')
    
      })

    test('Should render firstName and lastname - screen', () => {
       render(<MyAwesomeApp/>)
       screen.debug();

      //  console.log(container.innerHTML)
      // const h1 = screen.getByRole('heading', {
      //   level: 1,
      // })
      
      const h1 = screen.getByTestId('first-name-title');
      expect(h1.innerHTML).toContain('Raul')
      })

      // test('should match snapshot', () => {
      //   const { container } = render(<MyAwesomeApp />);

      //   expect(container).toMatchSnapshot();
      // })
})