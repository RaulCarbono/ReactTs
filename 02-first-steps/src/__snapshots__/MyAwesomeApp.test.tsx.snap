// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > should match snapshot 1`] = `
<div>
  <div>
    <h1
      data-testid="first-name-title"
    >
      Raul
      !!
    </h1>
    <h3
      class="estoy loco"
    >
      Carbono
    </h3>
  </div>
</div>
`;
